### PR TITLE
Issue #15: Make sure created CRs have names that are valid (and not too long)

### PR DESCRIPTION
**Pull request by sseago**
Use GenerateName instead of Name for created resources. Look up by label instead of using Get calls.

## Summary by CodeRabbit

* **Bug Fixes**
  * Reduced naming conflicts and races by switching to generated resource names and safer list-based lookups; cleanup now reliably removes related resources in bulk with per-resource logging.

* **Improvements**
  * More idempotent lifecycle handling across upload phases; generated-name prefixes and persisted name annotations improve cross-stage tracking and clearer logs.
  * Temporary volume handling improved to avoid duplicates.

* **Tests**
  * Expanded coverage for naming, labeling and idempotency scenarios.

### CODE REVIEW

**Comment by mpryc** [comment]
@sseago

```shell
$ oc oadp backup create --include-namespaces cirros-test-2 --resource-policies-configmap volume-policy --snapshot-move-data=true updated-long-name-backup-test-1

$ oc get pod kubevirt-datamover-controller-manager-5446b5cd6d-xzgvj 
NAME                                                     READY   STATUS    RESTARTS   AGE
kubevirt-datamover-controller-manager-5446b5cd6d-xzgvj   1/1     Running   0          2m44s

$ oc logs kubevirt-datamover-controller-manager-5446b5cd6d-xzgvj
[...]
2026-02-26T13:14:50Z	INFO	Reconciling DataUpload with kubevirt datamover	{"controller": "kubevirt-dataupload", "controllerGroup": "velero.io", "controllerKind": "DataUpload", "DataUpload": {"name":"du-updated-long-name-backup-test-1-cirros-test-cont-2-53148237","namespace":"openshift-adp"}, "namespace": "openshift-adp", "name": "du-updated-long-name-backup-test-1-cirros-test-cont-2-53148237", "reconcileID": "21628605-018a-4394-a05b-6ea6dd21b0ce", "dataUpload": {"name":"du-updated-long-name-backup-test-1-cirros-test-cont-2-53148237","namespace":"openshift-adp"}, "phase": "Accepted"}
2026-02-26T13:14:50Z	INFO	Handling Accepted phase DataUpload	{"controller": "kubevirt-dataupload", "controllerGroup": "velero.io", "controllerKind": "DataUpload", "DataUpload": {"name":"du-updated-long-name-backup-test-1-cirros-test-cont-2-53148237","namespace":"openshift-adp"}, "namespace": "openshift-adp", "name": "du-updated-long-name-backup-test-1-cirros-test-cont-2-53148237", "reconcileID": "21628605-018a-4394-a05b-6ea6dd21b0ce"}
2026-02-26T13:14:50Z	DEBUG	Temporary PVC already exists	{"controller": "kubevirt-dataupload", "controllerGroup": "velero.io", "controllerKind": "DataUpload", "DataUpload": {"name":"du-updated-long-name-backup-test-1-cirros-test-cont-2-53148237","namespace":"openshift-adp"}, "namespace": "openshift-adp", "name": "du-updated-long-name-backup-test-1-cirros-test-cont-2-53148237", "reconcileID": "21628605-018a-4394-a05b-6ea6dd21b0ce", "pvc": "kubevirt-backup-du-updated-long-name-backup-test-1-cirros-cd7xn"}
2026-02-26T13:14:50Z	INFO	Temporary PVC ready	{"controller": "kubevirt-dataupload", "controllerGroup": "velero.io", "controllerKind": "DataUpload", "DataUpload": {"name":"du-updated-long-name-backup-test-1-cirros-test-cont-2-53148237","namespace":"openshift-adp"}, "namespace": "openshift-adp", "name": "du-updated-long-name-backup-test-1-cirros-test-cont-2-53148237", "reconcileID": "21628605-018a-4394-a05b-6ea6dd21b0ce", "pvc": "kubevirt-backup-du-updated-long-name-backup-test-1-cirros-cd7xn"}
2026-02-26T13:14:50Z	DEBUG	VirtualMachineBackup already exists, skipping VMBT preparation	{"controller": "kubevirt-dataupload", "controllerGroup": "velero.io", "controllerKind": "DataUpload", "DataUpload": {"name":"du-updated-long-name-backup-test-1-cirros-test-cont-2-53148237","namespace":"openshift-adp"}, "namespace": "openshift-adp", "name": "du-updated-long-name-backup-test-1-cirros-test-cont-2-53148237", "reconcileID": "21628605-018a-4394-a05b-6ea6dd21b0ce", "vmb": "vmb-du-updated-long-name-backup-test-1-cirros-test-cont-2-wdp72"}
2026-02-26T13:14:50Z	INFO	VirtualMachineBackup in progress, requeuing	{"controller": "kubevirt-dataupload", "controllerGroup": "velero.io", "controllerKind": "DataUpload", "DataUpload": {"name":"du-updated-long-name-backup-test-1-cirros-test-cont-2-53148237","namespace":"openshift-adp"}, "namespace": "openshift-adp", "name": "du-updated-long-name-backup-test-1-cirros-test-cont-2-53148237", "reconcileID": "21628605-018a-4394-a05b-6ea6dd21b0ce"}
2026-02-26T13:14:55Z	INFO	Reconciling DataUpload with kubevirt datamover	{"controller": "kubevirt-dataupload", "controllerGroup": "velero.io", "controllerKind": "DataUpload", "DataUpload": {"name":"du-updated-long-name-backup-test-1-cirros-test-cont-2-53148237","namespace":"openshift-adp"}, "namespace": "openshift-adp", "name": "du-updated-long-name-backup-test-1-cirros-test-cont-2-53148237", "reconcileID": "d6781129-7f7e-439c-b2c6-abad40e14e33", "dataUpload": {"name":"du-updated-long-name-backup-test-1-cirros-test-cont-2-53148237","namespace":"openshift-adp"}, "phase": "Accepted"}
2026-02-26T13:14:55Z	INFO	Handling Accepted phase DataUpload	{"controller": "kubevirt-dataupload", "controllerGroup": "velero.io", "controllerKind": "DataUpload", "DataUpload": {"name":"du-updated-long-name-backup-test-1-cirros-test-cont-2-53148237","namespace":"openshift-adp"}, "namespace": "openshift-adp", "name": "du-updated-long-name-backup-test-1-cirros-test-cont-2-53148237", "reconcileID": "d6781129-7f7e-439c-b2c6-abad40e14e33"}
2026-02-26T13:14:55Z	DEBUG	Temporary PVC already exists	{"controller": "kubevirt-dataupload", "controllerGroup": "velero.io", "controllerKind": "DataUpload", "DataUpload": {"name":"du-updated-long-name-backup-test-1-cirros-test-cont-2-53148237","namespace":"openshift-adp"}, "namespace": "openshift-adp", "name": "du-updated-long-name-backup-test-1-cirros-test-cont-2-53148237", "reconcileID": "d6781129-7f7e-439c-b2c6-abad40e14e33", "pvc": "kubevirt-backup-du-updated-long-name-backup-test-1-cirros-cd7xn"}
2026-02-26T13:14:55Z	INFO	Temporary PVC ready	{"controller": "kubevirt-dataupload", "controllerGroup": "velero.io", "controllerKind": "DataUpload", "DataUpload": {"name":"du-updated-long-name-backup-test-1-cirros-test-cont-2-53148237","namespace":"openshift-adp"}, "namespace": "openshift-adp", "name": "du-updated-long-name-backup-test-1-cirros-test-cont-2-53148237", "reconcileID": "d6781129-7f7e-439c-b2c6-abad40e14e33", "pvc": "kubevirt-backup-du-updated-long-name-backup-test-1-cirros-cd7xn"}
2026-02-26T13:14:55Z	DEBUG	VirtualMachineBackup already exists, skipping VMBT preparation	{"controller": "kubevirt-dataupload", "controllerGroup": "velero.io", "controllerKind": "DataUpload", "DataUpload": {"name":"du-updated-long-name-backup-test-1-cirros-test-cont-2-53148237","namespace":"openshift-adp"}, "namespace": "openshift-adp", "name": "du-updated-long-name-backup-test-1-cirros-test-cont-2-53148237", "reconcileID": "d6781129-7f7e-439c-b2c6-abad40e14e33", "vmb": "vmb-du-updated-long-name-backup-test-1-cirros-test-cont-2-wdp72"}
2026-02-26T13:14:55Z	INFO	VirtualMachineBackup in progress, requeuing	{"controller": "kubevirt-dataupload", "controllerGroup": "velero.io", "controllerKind": "DataUpload", "DataUpload": {"name":"du-updated-long-name-backup-test-1-cirros-test-cont-2-53148237","namespace":"openshift-adp"}, "namespace": "openshift-adp", "name": "du-updated-long-name-backup-test-1-cirros-test-cont-2-53148237", "reconcileID": "d6781129-7f7e-439c-b2c6-abad40e14e33"}
2026-02-26T13:15:00Z	INFO	Reconciling DataUpload with kubevirt datamover	{"controller": "kubevirt-dataupload", "controllerGroup": "velero.io", "controllerKind": "DataUpload", "DataUpload": {"name":"du-updated-long-name-backup-test-1-cirros-test-cont-2-53148237","namespace":"openshift-adp"}, "namespace": "openshift-adp", "name": "du-updated-long-name-backup-test-1-cirros-test-cont-2-53148237", "reconcileID": "edb871f9-27da-4597-be1b-3427e9434c9f", "dataUpload": {"name":"du-updated-long-name-backup-test-1-cirros-test-cont-2-53148237","namespace":"openshift-adp"}, "phase": "Accepted"}
2026-02-26T13:15:00Z	INFO	Handling Accepted phase DataUpload	{"controller": "kubevirt-dataupload", "controllerGroup": "velero.io", "controllerKind": "DataUpload", "DataUpload": {"name":"du-updated-long-name-backup-test-1-cirros-test-cont-2-53148237","namespace":"openshift-adp"}, "namespace": "openshift-adp", "name": "du-updated-long-name-backup-test-1-cirros-test-cont-2-53148237", "reconcileID": "edb871f9-27da-4597-be1b-3427e9434c9f"}
2026-02-26T13:15:00Z	DEBUG	Temporary PVC already exists	{"controller": "kubevirt-dataupload", "controllerGroup": "velero.io", "controllerKind": "DataUpload", "DataUpload": {"name":"du-updated-long-name-backup-test-1-cirros-test-cont-2-53148237","namespace":"openshift-adp"}, "namespace": "openshift-adp", "name": "du-updated-long-name-backup-test-1-cirros-test-cont-2-53148237", "reconcileID": "edb871f9-27da-4597-be1b-3427e9434c9f", "pvc": "kubevirt-backup-du-updated-long-name-backup-test-1-cirros-cd7xn"}
2026-02-26T13:15:00Z	INFO	Temporary PVC ready	{"controller": "kubevirt-dataupload", "controllerGroup": "velero.io", "controllerKind": "DataUpload", "DataUpload": {"name":"du-updated-long-name-backup-test-1-cirros-test-cont-2-53148237","namespace":"openshift-adp"}, "namespace": "openshift-adp", "name": "du-updated-long-name-backup-test-1-cirros-test-cont-2-53148237", "reconcileID": "edb871f9-27da-4597-be1b-3427e9434c9f", "pvc": "kubevirt-backup-du-updated-long-name-backup-test-1-cirros-cd7xn"}
2026-02-26T13:15:00Z	DEBUG	VirtualMachineBackup already exists, skipping VMBT preparation	{"controller": "kubevirt-dataupload", "controllerGroup": "velero.io", "controllerKind": "DataUpload", "DataUpload": {"name":"du-updated-long-name-backup-test-1-cirros-test-cont-2-53148237","namespace":"openshift-adp"}, "namespace": "openshift-adp", "name": "du-updated-long-name-backup-test-1-cirros-test-cont-2-53148237", "reconcileID": "edb871f9-27da-4597-be1b-3427e9434c9f", "vmb": "vmb-du-updated-long-name-backup-test-1-cirros-test-cont-2-wdp72"}
2026-02-26T13:15:00Z	INFO	VirtualMachineBackup in progress, requeuing	{"controller": "kubevirt-dataupload", "controllerGroup": "velero.io", "controllerKind": "DataUpload", "DataUpload": {"name":"du-updated-long-name-backup-test-1-cirros-test-cont-2-53148237","namespace":"openshift-adp"}, "namespace": "openshift-adp", "name": "du-updated-long-name-backup-test-1-cirros-test-cont-2-53148237", "reconcileID": "edb871f9-27da-4597-be1b-3427e9434c9f"}
```

The VMB object created:
```shell
$ oc get vmbackuptracker -n cirros-test-2 -o yaml
apiVersion: v1
items:
- apiVersion: backup.kubevirt.io/v1alpha1
  kind: VirtualMachineBackupTracker
  metadata:
    creationTimestamp: "2026-02-26T13:12:49Z"
    generateName: vmbt-cirros-test-cont-2-
    generation: 1
    labels:
      kubevirt-datamover.io/vm-name: cirros-test-cont-2
      velero.io/dataupload-name: du-updated-long-name-backup-test-1-cirros-test-cont-2-53148237
    name: vmbt-cirros-test-cont-2-sccmq
    namespace: cirros-test-2
    resourceVersion: "56277207"
    uid: ef952287-3ed0-4b62-be79-597b07329d51
  spec:
    source:
      apiGroup: kubevirt.io
      kind: VirtualMachine
      name: cirros-test-cont-2
kind: List
metadata:
  resourceVersion: ""
```

OC events:
```shell
$ oc get events -n cirros-test-2 --sort-by='.lastTimestamp' 2>&1 | tail -30
LAST SEEN   TYPE      REASON                  OBJECT                                                                                  MESSAGE
24m         Normal    Provisioning            persistentvolumeclaim/kubevirt-backup-du-updated-long-name-backup-test-1-cirros-cd7xn   External provisioner is provisioning volume for claim "cirros-test-2/kubevirt-backup-du-updated-long-name-backup-test-1-cirros-cd7xn"
24m         Normal    ExternalProvisioning    persistentvolumeclaim/kubevirt-backup-du-updated-long-name-backup-test-1-cirros-cd7xn   Waiting for a volume to be created either by the external provisioner 'openshift-storage.rbd.csi.ceph.com' or manually by the system administrator. If volume creation is delayed, please verify that the provisioner is running and correctly registered.
24m         Normal    ProvisioningSucceeded   persistentvolumeclaim/kubevirt-backup-du-updated-long-name-backup-test-1-cirros-cd7xn   Successfully provisioned volume pvc-e638e947-5f47-47ea-9916-370c6391bdb1
24m         Warning   FailedCreate            virtualmachineinstance/cirros-test-cont-2                                               Error creating attachment pod: Pod "hp-volume-cjfvz" is invalid: [spec.volumes[1].name: Invalid value: "vmb-du-updated-long-name-backup-test-1-cirros-test-cont-2-wdp72-backup-target-pvc": must be no more than 63 characters, spec.containers[0].volumeMounts[1].name: Not found: "vmb-du-updated-long-name-backup-test-1-cirros-test-cont-2-wdp72-backup-target-pvc"]
24m         Warning   FailedCreate            virtualmachineinstance/cirros-test-cont-2                                               Error creating attachment pod: Pod "hp-volume-xq49f" is invalid: [spec.volumes[1].name: Invalid value: "vmb-du-updated-long-name-backup-test-1-cirros-test-cont-2-wdp72-backup-target-pvc": must be no more than 63 characters, spec.containers[0].volumeMounts[1].name: Not found: "vmb-du-updated-long-name-backup-test-1-cirros-test-cont-2-wdp72-backup-target-pvc"]
24m         Warning   FailedCreate            virtualmachineinstance/cirros-test-cont-2                                               Error creating attachment pod: Pod "hp-volume-vbdzh" is invalid: [spec.volumes[1].name: Invalid value: "vmb-du-updated-long-name-backup-test-1-cirros-test-cont-2-wdp72-backup-target-pvc": must be no more than 63 characters, spec.containers[0].volumeMounts[1].name: Not found: "vmb-du-updated-long-name-backup-test-1-cirros-test-cont-2-wdp72-backup-target-pvc"]
24m         Warning   FailedCreate            virtualmachineinstance/cirros-test-cont-2                                               Error creating attachment pod: Pod "hp-volume-4gtd7" is invalid: [spec.volumes[1].name: Invalid value: "vmb-du-updated-long-name-backup-test-1-cirros-test-cont-2-wdp72-backup-target-pvc": must be no more than 63 characters, spec.containers[0].volumeMounts[1].name: Not found: "vmb-du-updated-long-name-backup-test-1-cirros-test-cont-2-wdp72-backup-target-pvc"]
24m         Warning   FailedCreate            virtualmachineinstance/cirros-test-cont-2                                               Error creating attachment pod: Pod "hp-volume-b8zn4" is invalid: [spec.volumes[1].name: Invalid value: "vmb-du-updated-long-name-backup-test-1-cirros-test-cont-2-wdp72-backup-target-pvc": must be no more than 63 characters, spec.containers[0].volumeMounts[1].name: Not found: "vmb-du-updated-long-name-backup-test-1-cirros-test-cont-2-wdp72-backup-target-pvc"]
24m         Warning   FailedCreate            virtualmachineinstance/cirros-test-cont-2                                               Error creating attachment pod: Pod "hp-volume-6bvws" is invalid: [spec.volumes[1].name: Invalid value: "vmb-du-updated-long-name-backup-test-1-cirros-test-cont-2-wdp72-backup-target-pvc": must be no more than 63 characters, spec.containers[0].volumeMounts[1].name: Not found: "vmb-du-updated-long-name-backup-test-1-cirros-test-cont-2-wdp72-backup-target-pvc"]
24m         Warning   FailedCreate            virtualmachineinstance/cirros-test-cont-2                                               Error creating attachment pod: Pod "hp-volume-dd945" is invalid: [spec.volumes[1].name: Invalid value: "vmb-du-updated-long-name-backup-test-1-cirros-test-cont-2-wdp72-backup-target-pvc": must be no more than 63 characters, spec.containers[0].volumeMounts[1].name: Not found: "vmb-du-updated-long-name-backup-test-1-cirros-test-cont-2-wdp72-backup-target-pvc"]
24m         Warning   FailedCreate            virtualmachineinstance/cirros-test-cont-2                                               Error creating attachment pod: Pod "hp-volume-bp5ld" is invalid: [spec.volumes[1].name: Invalid value: "vmb-du-updated-long-name-backup-test-1-cirros-test-cont-2-wdp72-backup-target-pvc": must be no more than 63 characters, spec.containers[0].volumeMounts[1].name: Not found: "vmb-du-updated-long-name-backup-test-1-cirros-test-cont-2-wdp72-backup-target-pvc"]
24m         Warning   FailedCreate            virtualmachineinstance/cirros-test-cont-2                                               Error creating attachment pod: Pod "hp-volume-lf6vh" is invalid: [spec.volumes[1].name: Invalid value: "vmb-du-updated-long-name-backup-test-1-cirros-test-cont-2-wdp72-backup-target-pvc": must be no more than 63 characters, spec.containers[0].volumeMounts[1].name: Not found: "vmb-du-updated-long-name-backup-test-1-cirros-test-cont-2-wdp72-backup-target-pvc"]
13m         Warning   FailedCreate            virtualmachineinstance/cirros-test-cont-2                                               (combined from similar events): Error creating attachment pod: Pod "hp-volume-tgpqf" is invalid: [spec.volumes[1].name: Invalid value: "vmb-du-updated-long-name-backup-test-1-cirros-test-cont-2-wdp72-backup-target-pvc": must be no more than 63 characters, spec.containers[0].volumeMounts[1].name: Not found: "vmb-du-updated-long-name-backup-test-1-cirros-test-cont-2-wdp72-backup-target-pvc"]
4m51s       Warning   Migrated                virtualmachineinstance/cirros-test-cont-2                                               EvictionStrategy is set but vmi is not migratable; cannot migrate VMI: PVC cirros-test-cont-2-dv is not shared, live migration requires that all PVCs must be shared (using ReadWriteMany access mode)
2m25s       Warning   FailedCreate            virtualmachineinstance/cirros-test-cont-2                                               Error creating attachment pod: Pod "hp-volume-n4k2v" is invalid: [spec.volumes[1].name: Invalid value: "vmb-du-updated-long-name-backup-test-1-cirros-test-cont-2-wdp72-backup-target-pvc": must be no more than 63 characters, spec.containers[0].volumeMounts[1].name: Not found: "vmb-du-updated-long-name-backup-test-1-cirros-test-cont-2-wdp72-backup-target-pvc"]

```

# Root cause
KubeVirt uses <vmb-name>-backup-target-pvc as a volume name in the hotplug attachment pod. The VMB name is 63 chars + -backup-target-pvc (18 chars) = 81 chars, which exceeds the 63-char limit for pod volume names. So the name length bug wasn't fully fixed - GenerateName caps the VMB name itself at 63, but KubeVirt appends a suffix to it for the hotplug volume, which overflows.

# AI prompt to fix:
```shell
Fix VMB name length bug causing KubeVirt hotplug volume attachment failure.

## Root Cause

KubeVirt internally constructs a hotplug volume name by appending "-backup-target-pvc"
(18 chars) to the VirtualMachineBackup resource name. Pod volume names are DNS labels
and must be <= 63 characters. Our controller creates VMB resources using GenerateName
which K8s silently truncates to produce names up to 63 characters. With the 18-char
suffix appended by KubeVirt, the resulting volume name can be up to 81 characters,
which K8s rejects when creating the hotplug attachment pod:

  Error creating attachment pod: Pod "hp-volume-cjfvz" is invalid:
  spec.volumes[1].name: Invalid value:
  "vmb-du-updated-long-name-backup-test-1-cirros-test-cont-2-wdp72-backup-target-pvc":
  must be no more than 63 characters

This causes the VMB to get stuck in Initializing phase forever. The controller sees
no "Done" condition and keeps requeuing with "VirtualMachineBackup in progress".

## What to Fix

File: internal/controller/kubevirt_dataupload_controller.go

1. Add constants in the const block (around line 69, after the AnnotationVMBTName line):

   // k8sGenerateNameRandomLen is the number of random characters K8s appends to GenerateName
   k8sGenerateNameRandomLen = 5

   // maxVMBNameLen is the max allowed VMB name length. KubeVirt constructs a
   // hotplug volume name as "<vmb-name>-backup-target-pvc" which must be a
   // valid DNS label (≤ 63 chars). Reserve 18 chars for that suffix.
   maxVMBNameLen = 63 - len("-backup-target-pvc") // = 45

2. Add a helper function (near the other helper functions like getVeleroBackupName):

   // safeGenerateNamePrefix truncates a GenerateName prefix so that the final
   // name (prefix + 5 random chars) does not exceed maxNameLen.
   func safeGenerateNamePrefix(prefix string, maxNameLen int) string {
       maxPrefix := maxNameLen - k8sGenerateNameRandomLen
       if maxPrefix < 1 {
           maxPrefix = 1
       }
       if len(prefix) > maxPrefix {
           prefix = prefix[:maxPrefix]
       }
       return prefix
   }

3. In the ensureVMBackup function, change this line:
     GenerateName: fmt.Sprintf("vmb-%s-", du.Name),
   To:
     GenerateName: safeGenerateNamePrefix(fmt.Sprintf("vmb-%s-", du.Name), maxVMBNameLen),

   This ensures VMB names are at most 45 chars, leaving room for KubeVirt's
   18-char "-backup-target-pvc" suffix.

4. Update tests in kubevirt_dataupload_controller_test.go:
   - Any test that creates VMB fixtures with names derived from long DataUpload names
     must use names that are <= 45 chars total (matching the new truncation behavior).
   - Add a unit test for safeGenerateNamePrefix verifying:
     - Short prefixes are returned unchanged
     - Long prefixes are truncated so prefix + 5 random chars = maxNameLen
     - Edge cases (maxNameLen <= 5) produce a prefix of at least 1 char

5. After making all changes, run: make lint-fix && make test
   All existing tests and new tests must pass.

## Key Design Details

- VMB resources are looked up by label selectors (LabelDataUploadUID), NOT by name,
  so truncating the GenerateName prefix does not break any lookups.
- The labels on the VMB still store the full du.Name (up to 62 chars) and du.UID
  (36 chars), both within the 63-char label value limit. Labels are NOT truncated.
- Other resources (PVC, VMBT, Pod) also use GenerateName but don't have downstream
  suffix issues from KubeVirt, so they don't strictly need this fix. However, you may
  apply safeGenerateNamePrefix to them defensively if you wish.
- K8s GenerateName appends exactly 5 random alphanumeric characters to the prefix.
  K8s itself silently truncates prefixes to 58 chars (63 - 5). Our fix needs a
  TIGHTER limit: 40 char prefix (45 - 5) because of KubeVirt's additional suffix.

```

**Review by mpryc — CHANGES_REQUESTED** [review]
As in my previous comment.

**Comment by sseago** [comment]
@mpryc latest version puts du name in annotation, so it should now be  compatible with long DU names.

File: internal/controller/kubevirt_dataupload_controller.go [hunk 1]
```diff
@@ -643,25 +676,23 @@ func (r *KubeVirtDataUploadReconciler) handleInProgress(ctx context.Context, log
...
-			if err := r.updatePhase(ctx, du, velerov2alpha1.DataUploadPhaseFailed, "Datamover pod not found"); err != nil {
-				return ctrl.Result{}, err
-			}
-			return ctrl.Result{}, nil
-		}
 		return ctrl.Result{}, fmt.Errorf("failed to get datamover pod: %w", err)
 	}
+	if pod == nil {
+		// Pod not found - this is unexpected in InProgress phase
+		logger.Error(err, "Datamover pod not found", "dataUpload", du.Name, "namespace", podNamespace)
```
Comment (shubham-pampattiwar) [inline internal/controller/kubevirt_dataupload_controller.go]: Nit: err is guaranteed to be nil here (the non-nil case returns above). Consider using `logger.Error(nil, ...)` explicitly or passing a real error like `fmt.Errorf("datamover pod not found for DataUpload %s", du.Name)`so the structured log includes a meaningful error field.

**Review by shubham-pampattiwar — COMMENTED** [review]
Label value length:
Kubernetes label values are capped at 63 characters. `LabelDataUploadName: du.Name` is set as a label on PVC, VMB, VMBT, and Pod. With long backup names, `du.Name` can easily exceed 63 chars (e.g. `du-ultra-very-long-backup-name-that-is-over-sixty-three-characters-cirros-test-cont-2-a9ed8aa7` is 90 chars). The `r.Create()` calls will be rejected by the API server.

The good news is this PR already uses `LabelDataUploadUID` (UUID, always 36 chars) for all List lookups, so the lookup logic is safe. The problem is `LabelDataUploadName` is still set as a label. Similarly, the new `vmbtLabelVMName` label uses the VM name which could also exceed 63 chars.

Proposed approach:

Use labels only for values guaranteed ≤ 63 chars, and annotations for full-length traceability:

```go
Labels (for lookups):
  LabelDataUploadUID:  du.UID          // 36 chars, always safe
  vmbtLabelVMName:     sha256(vm)[:16] // fixed length, for VMBT grouping

Annotations (for traceability):
  AnnotationDataUploadName: du.Name       // full name, no limit
  AnnotationVMName:         vmName        // full name, no limit
  AnnotationBackupName:     backupName    // full name, no limit
```

This same pattern needs to cascade to the plugin code as well.

File: internal/controller/kubevirt_dataupload_controller.go [hunk 2]
```diff
@@ -65,6 +65,19 @@ const (
 
 	// defaultCredentialKey is the default key in BSL credential secrets
 	defaultCredentialKey = "cloud"
+
+	// vmbtLabelVMName is the label key for the VM name on a VirtualMachineBackupTracker
+	vmbtLabelVMName = "kubevirt-datamover.io/vm-name"
+	// AnnotationVMBTName is the annotation key for the generated VMBT name on a DataUpload
+	AnnotationVMBTName = "kubevirt-datamover.io/vmbt-name"
```
Comment (kaovilai) [inline internal/controller/kubevirt_dataupload_controller.go:72]: do we want to be consistent with other repos? oadp.openshift.io? https://github.com/openshift/oadp-operator/blob/87b9ef4d3715fefcff8353b4bbb32de521eec971/config/non-admin-controller_rbac/nonadmindownloadrequest_viewer_role.yaml#L17

**Comment by sseago** [comment]
> Label value length: Kubernetes label values are capped at 63 characters. `LabelDataUploadName: du.Name` is set as a label on PVC, VMB, VMBT, and Pod. With long backup names, `du.Name` can easily exceed 63 chars (e.g. `du-ultra-very-long-backup-name-that-is-over-sixty-three-characters-cirros-test-cont-2-a9ed8aa7` is 90 chars). The `r.Create()` calls will be rejected by the API server.
> 
> The good news is this PR already uses `LabelDataUploadUID` (UUID, always 36 chars) for all List lookups, so the lookup logic is safe. The problem is `LabelDataUploadName` is still set as a label. Similarly, the new `vmbtLabelVMName` label uses the VM name which could also exceed 63 chars.
> 
> Proposed approach:
> 
> Use labels only for values guaranteed ≤ 63 chars, and annotations for full-length traceability:
> 
> ```go
> Labels (for lookups):
>   LabelDataUploadUID:  du.UID          // 36 chars, always safe
>   vmbtLabelVMName:     sha256(vm)[:16] // fixed length, for VMBT grouping
> 
> Annotations (for traceability):
>   AnnotationDataUploadName: du.Name       // full name, no limit
>   AnnotationVMName:         vmName        // full name, no limit
>   AnnotationBackupName:     backupName    // full name, no limit
> ```
> 
> This same pattern needs to cascade to the plugin code as well.

@shubham-pampattiwar The latest version of this PR moves this to an annotation.

**Review by mpryc — COMMENTED** [review]
/lgtm

File: internal/controller/kubevirt_dataupload_controller.go (hunk 1 above)
Comment (sseago) [reply internal/controller/kubevirt_dataupload_controller.go]: Is there any value in passing fmt.Errorf() output into logger.Error with the same message we're already passing in above? I could also just explicitly pass nil into logger.Error here.

File: internal/controller/kubevirt_dataupload_controller.go (hunk 2 above)
Comment (sseago) [reply internal/controller/kubevirt_dataupload_controller.go:72]: @kaovilai maybe we do, but this PR is moving around existing strings. If we want to revisit our naming convention, that should be a separate PR (and coordinated with a similar plugin PR).

**Review by shubham-pampattiwar — DISMISSED** [review]

**Review by shubham-pampattiwar — APPROVED** [review]

**Review by mpryc — APPROVED** [review]

**Review by mpryc — APPROVED** [review]

**Review by shubham-pampattiwar — APPROVED** [review]
